Task 7/Task 7.2/script.js

```diff
@@ -0,0 +1,115 @@
+calculateExpression();
+
+function calculateExpression () {
+    try {
+        let userInput = document.getElementById("userExpression");
+        let output = document.getElementById("result");
+        userInput.addEventListener("blur", (event) => {
+            let result = calculator (event.target.value);
+            output.innerHTML = "Your result is " + result;
+        });
+    } catch (e) {
+        alert(e.message);
+    }
+}
+
+function calculator (userExpression) {
+    try {
+        userExpression = checkExpression(userExpression);
+        let ariphmeticSigns = getAriphmeticSigns(userExpression);
+        let digitsArray = getDigits(userExpression);
+        let result = getResult (digitsArray, ariphmeticSigns);
+        return result;
+    } catch (e) {
+        throw e;
+    }
+}
+
+/* This function checks if format of user's expression is correct or not. */
+
+function checkExpression (expression) {
+    expression = cutRightPartOfExpression(expression.trim());
+    if (expression.startsWith("*") || expression.startsWith("/")) {
+        throw new Error ("Your expression isn't valid.");
+    } else {
+        const regexp = /[0-9]/;
+        const symbolsRegexp = /[A-Za-zА-Яа-яЁё]/;
+        let index = expression.search(regexp);
+        let symbolsIndex = expression.search(symbolsRegexp);
+        if (index == -1 || index > 1 || symbolsIndex != -1) {
+            throw new Error ("Your expression isn't valid.");
+        }
+        return expression;
+    }
+}
+
+/* This method cuts everything that is located after sign "=" */
+
+function cutRightPartOfExpression (expression) {
+    let index = expression.indexOf("=");
+    let result = expression.slice(0, index + 1);
+    return result;
+}
+
+function getAriphmeticSigns(expression) {
+    let signs = [];
+    const separators = ["+", "-", "/", "*"];
+    for (let i = 0; i < expression.length; i ++) {
+        if (separators.includes(expression[i])) {
+            signs.push(expression[i]);
+        }
+    }
```

Comment:
```js
expression.filter(char => separators.includes(char))
``` 
(if expression is an Array) is better than `for`. You can either convert string to an array with `Array.from` or steal `filter` from Array prototype and call it on a string with `<function>.call()` (but that is an advanced technique, you'll learn it later)